### ISSUE TITLE
FL0034: False positive with F# 5

### ISSUE THREAD

**Issue by Evangelink** [fsprojects/FSharpLint#469]
### Description

When installing F# 5, I have faced an error due to an overload resolution change and after fixing it, I keep getting the `FL0034`. See https://github.com/dotnet/fsharp/issues/10424

### Repro steps

Before F# 5:

```fsharp
let frameDuration =
	getFrameRate model
	|> Option.defaultValue(10</s>)
	|> fun rate -> 1 / rate * 1000<ms/s>
	|> int
Cmd.batch [
	Cmd.ofMsg NextFrame
	Cmd.OfAsync.perform Async.Sleep frameDuration (fun () -> AutoPlayNextFrame)
]
```

With F# 5 (a new `TimeSpan` overload as been introduced)

```fsharp
let frameDuration =
	getFrameRate model
	|> Option.defaultValue(10</s>)
	|> fun rate -> 1 / rate * 1000<ms/s>
	|> int
Cmd.batch [
	Cmd.ofMsg NextFrame
	Cmd.OfAsync.perform (fun (x:int) -> Async.Sleep x) frameDuration (fun () -> AutoPlayNextFrame)
]
```

### Expected behavior

No warning on the `(fun (x:int) -> Async.Sleep x)`

### Actual behavior

Warning

### Known workarounds

None, to my knowledge.

### Related information

* Operating system: Win10
* net48, F# 5
* VS2019 latest and VS2019 Preview
